internal/release/promote/promotion.go

```diff
@@ -195,18 +197,46 @@ func (p *Promotion) Promote(opts Opts) (string, error) {
 		return p.perform(performParams)
 	}
 
-	// Prompt user to select creating a pull request
-	answer, err := p.PromptProvider.SelectCreatingPromotionPullRequest()
-	if err != nil {
-		return "", fmt.Errorf("selecting create promotion pull request: %w", err)
-	}
+loop:
+	for {
+		// Prompt user to select creating a pull request
+		action, err := p.PromptProvider.SelectPromotionAction()
+		if err != nil {
+			return "", fmt.Errorf("selecting create promotion pull request: %w", err)
+		}
 
-	switch answer {
-	case Draft:
-		performParams.draft = true
-	case Cancel:
-		p.PromptProvider.PrintCanceled()
-		return "", nil
+		switch action {
+		case CreatePR:
+			break loop
+		case CreateDraft:
+			performParams.draft = true
+			break loop
+		case ViewGitLog:
+			for _, item := range selectedList.Items {
+				p.println(color.InBold(fmt.Sprintf("--- %s", item.Name)))
+				source, target := at(item.Releases, sourceEnvIndex), at(item.Releases, targetEnvIndex)
+
+				info, err := getReleaseInfo(source, target, performParams)
+				if err != nil {
+					p.printf("error getting release information %s: %v\n", item.Name, err)
+					continue
+				}
+
+				for _, commit := range info.Commits {
+					p.printf("%s %s %s\n", color.InYellow(commit.ShortSha), color.InCyan(commit.GitHubAuthor), commit.ShortMessage)
```

Comment:
`style`?? 🙉

Reply:
Hot take. We abuse style. I have no idea which styles these should be, nor if I need to create those styles, what styles I should name them. There comes a point when "style" falls apart.